### PR TITLE
NSPR-9557 Eliminate COFIG_SOFTKERNEL ert

### PR DESCRIPTION
**Pull request by larry9523**
This PR addresses the following issue on U30
1) Remove the attached PS (soft) kernel binary in ert CONFIG_SOFTKERNEL command 
2) One XCLBIN will only send one CONFIG_SOFTKERNEL command to PS and it will contain all SCU's metadata.
3) PS kernel binaries will be transferred to PS side via mgmt PF during load xclbin and will be activated when receiving CONFIG_SOFTKERNEL command
4) When PS kernel crashes due to bad PS kernel binary or bad host code, the host ert command will not hang but returns error
5) The race condition to write to ERT command queue between host mb_scheduler and ert scheduler is removed

This is an incompatible changes which requires host and device (U30 shell) update together. 
1) If new host XRT is talking to old shell, the load xclbin function will fail with following message
    "download_axlf: Platform does not support load xclbin"
    "icap_peer_xclbin_download: peer xclbin download err: -524"
    "ffff8800b7d43098 xocl_read_axlf_helper: Failed to download xclbin, err: -524"

2) If old host XRT is talking to new shell (you are not likely getting into this because once update shell, you are supposed to update host package), the PS (soft) kernel will not work properly because the PS kernel binary (inside CONFIG_SOFTKERNEL command) will not be accepted.

3) There will be a noticeable delay when load_xclbin because PS kernel binaries are now transferred to PS through a secure channel at mgmt side other than DMA at user PF. But for same xclbin uuid, this transfer is only required once. 

Reviewers, you are welcome to review the whole code changes. Particularly, please take a close look at the following portion for each reviewer
@xuhz mb_scheduler changes and any impact to the U30 reset
@houlz0507 fdt related changes
@xdavidz fdt changes, xrt_xclbin.[ch], zocl_xclbin[ch], impact to vck5000
@mamin506 zocl, sk daemon changes, impact to new kds

### CODE REVIEW

**Review by mamin506 — APPROVED** [review]
Overall this change look good to me.

Recently, there are a lot of soft kernel related change in mb_scheduler. I would port all of them into new kds.

**Review by houlz0507 — APPROVED** [review]

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c [hunk 1]
```diff
@@ -129,6 +130,72 @@ zocl_load_bitstream(struct drm_zocl_dev *zdev, char *buffer, int length)
...
+		sip->si_start = scu_idx;
+		sip->si_end = scu_idx + sp->m_num_instances - 1;
+
+		sip->si_bo = zocl_drm_create_bo(zdev->ddev, sp->m_image_size,
+		    ZOCL_BO_FLAGS_CMA);
+		if (IS_ERR(sip->si_bo)) {
+			ret = PTR_ERR(sip->si_bo);
+			DRM_ERROR("%s Failed to allocate BO: %d\n",
+			    __func__, ret);
+			return ret;
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:180]: mutex_unlock?

Comment (larry9523) [reply src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:181]: Good catch!

**Review by xdavidz — APPROVED** [review]

**Comment by abalasa** [comment]
retest this please

**Comment by abalasa** [comment]
retest this please

**Comment by abalasa** [comment]
retest this please

**Comment by dayeh-xilinx** [comment]
retest this please

**Comment by larry9523** [comment]
retest this please

**Review by xuhz — APPROVED** [review]

**Review by xuhz — COMMENTED** [review]
xocl_xclbin_download() is called only for baremetal case.
for cloud case, xocl_icap_download_axlf() is called, which means the soft kernel will not be downloaded

**Comment by larry9523** [comment]
retest this please

**Pull request merged**